components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth/action/execution/PreIssueAccessTokenResponseProcessorTest.java

```diff
@@ -0,0 +1,228 @@
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+import static org.wso2.carbon.identity.oauth.action.execution.PreIssueAccessTokenRequestBuilder.CLAIMS_PATH_PREFIX;
+import static org.wso2.carbon.identity.oauth.action.execution.PreIssueAccessTokenRequestBuilder.SCOPES_PATH_PREFIX;
+
+public class PreIssueAccessTokenResponseProcessorTest {
+
+    private static final String TAIL_CHARACTER = "-";
+    private AccessToken.Builder requestAccessTokenBuilder;
+    private MockedStatic<LoggerUtils> loggerUtils;
+
+    String original_iss = "https://localhost:9443/oauth2/token";
+    String original_sub = "3da43e1c-4087-46e2-ad46-08ccc76bf616";
+    String original_aud = "k58gg864hKaeLet9v7HkrFbhqsa";
+    String original_expire_in = "1742553132";
+    String original_clientId = "7k58gg864hKaeLet9v7HkrFbhqsa";
+    String original_applicationUser = "APPLICATION_USER";
+
+    @BeforeClass
+    public void init() {
+
+        loggerUtils = mockStatic(LoggerUtils.class);
+        loggerUtils.when(LoggerUtils::isDiagnosticLogsEnabled).thenReturn(true);
+
+        requestAccessTokenBuilder = new AccessToken.Builder()
+                .tokenType("JWT")
+                .addClaim(AccessToken.ClaimNames.ISS.getName(), original_iss)
+                .addClaim(AccessToken.ClaimNames.SUB.getName(), original_sub)
+                .addClaim(AccessToken.ClaimNames.AUD.getName(),
+                        List.of(original_aud))
+                .addClaim(AccessToken.ClaimNames.EXPIRES_IN.getName(), original_expire_in)
+                .addClaim(AccessToken.ClaimNames.CLIENT_ID.getName(), original_clientId)
+                .addClaim(AccessToken.ClaimNames.AUTHORIZED_USER_TYPE.getName(), original_applicationUser)
+                .addScope("openid")
+                .addScope("email")
+                .addScope("profile");
+    }
+
+    @Test
+    void testProcessSuccessResponse_WithInvalidScopePath() throws ActionExecutionResponseProcessorException {
+
+        List<PerformableOperation> operationsToPerform = new ArrayList<>();
+        List<String> customScopes = Arrays.asList("abc", "def");
+
+        PerformableOperation addOpScopeAsClaims = new PerformableOperation();
+        addOpScopeAsClaims.setOp(Operation.ADD);
+        addOpScopeAsClaims.setPath(CLAIMS_PATH_PREFIX + TAIL_CHARACTER);
+        addOpScopeAsClaims.setValue(new AccessToken.Claim("scope", customScopes));
+        operationsToPerform.add(addOpScopeAsClaims);
+
+        OAuthTokenReqMessageContext updatedAuthTokenReqMessageContext =
+                executeProcessSuccessResponse(operationsToPerform);
+        assertFalse(
+                CollectionUtils.containsAny(Arrays.asList(updatedAuthTokenReqMessageContext.getScope()), customScopes));
+        assertEquals(updatedAuthTokenReqMessageContext.getScope(), new String[]{"openid", "email", "profile"});
+        assertNull(updatedAuthTokenReqMessageContext.getAdditionalAccessTokenClaims().get("scope"));
+    }
```

Comment:
So this means we are not failing rather ignoring when scopes are sent in wrong path. This is fine but this experience should be consistent across extensions. Let's just check this a bit. I recall in authenticator we are failing in some validations